nf_core/refgenie.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+"""
+Update a nextflow.config file with refgenie genomes
+"""
+
+import logging
+import os
+import re
+
+# import refgenconf
+from warnings import warn
+
+import rich
+from rich.logging import RichHandler
+
+import nf_core.utils
+
+# Set up logging
+log = logging.getLogger(__name__)
+log.setLevel(logging.INFO)
+
+# # Setup rich traceback
+stderr = rich.console.Console(stderr=True, force_terminal=nf_core.utils.rich_force_colors())
+rich.traceback.install(console=stderr, width=200, word_wrap=True, extra_lines=1)
+
+NF_CFG_TEMPLATE = """
+// This is a read-only config file managed by refgenie. Manual changes to this file will be overwritten
+// To make changes here, use refgenie to update the reference genome data
+params {{
+  genomes {{
+{content}
+  }}
+}}
+"""
+
+
+def _print_nf_config(rgc):
+    """
+    Generate a nextflow config file with the genomes
+    from the refgenie config file
+    Adapted from: https://github.com/refgenie/refgenie_nfcore
+
+    Takes a RefGenConf object as argument
+    """
+    abg = rgc.list_assets_by_genome()
+    genomes_str = ""
+    for genome, asset_list in abg.items():
+        genomes_str += "    '{}' {{\n".format(genome)
+        for asset in asset_list:
+            try:
+                pth = rgc.seek(genome, asset)
+            # Catch general exception instead of refgencof exception --> no refgenconf import needed
+            except Exception as e:
+                log.warn(f"{genome}/{asset} is incomplete, ignoring...")
+            else:
+                genomes_str += '      {} = "{}"\n'.format(asset.ljust(20, " "), pth)
+
+    return NF_CFG_TEMPLATE.format(content=genomes_str)
+
+
+def _update_nextflow_home_config(refgenie_genomes_config_file, nxf_home):
+    """
+    Update the $NXF_HOME/config file by adding a includeConfig statement to it
+    for the 'refgenie_genomes_config_file' if not already defined
+    """
+    # Check if NXF_HOME/config exists and has a
+    include_config_string = f"includeConfig '{os.path.abspath(refgenie_genomes_config_file)}'\n"
```

Comment:
It would be nice to include at least some comment lines around this explaining what it is. For example (stealing comment style from Conda):

```suggestion
    include_config_string = textwrap.dedent(f"""
    ///// >>> nf-core + RefGenie >>> /////
    // !! Contents within this block are managed by 'nf-core/tools' !!
    // Includes auto-generated config file with RefGenie genome assets
    includeConfig '{os.path.abspath(refgenie_genomes_config_file)}'
    ///// <<< nf-core + RefGenie <<< /////
    """)
```

(needs `textwrap.dedent` import)